### ISSUE TITLE
Shard doesn't show in the list when it switched to degraded mode

### ISSUE THREAD

**Issue by anikeev-yadro** [nspcc-dev/neofs-node#1858]
## Expected Behavior
```neofs-cli control shards list``` should show all shards in all modes

## Current Behavior
```neofs-cli control shards list``` doesn't show shards in degraded mode

## Steps to Reproduce (for bugs)
1.Make some READ disk errors on the shard, then shard swithed to degraded mode:
```
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.096Z        warn        engine/engine.go:44        could not get object from shard        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "error count": 30, "error": "object ID is not set"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.097Z        debug        blobstor/control.go:49        closing...        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.099Z        debug        blobstor/control.go:12        opening...        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.100Z        debug        blobstor/control.go:32        initializing...        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.100Z        info        engine/engine.go:60        shard is moved in degraded mode due to error threshold        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "error count": 30}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.102Z        debug        shard/get.go:104        object is missing in write-cache        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.110Z        warn        engine/engine.go:44        could not get object from shard        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "error count": 31, "error": "object ID is not set"}
Oct 07 16:43:33 az neofs-node[19293]: 2022-10-07T16:43:33.110Z        info        engine/engine.go:60        shard is moved in degraded mode due to error threshold        {"shard_id": "Jj5YJnPQTXHSoYg6KpUuQi", "error count": 31}
```
2. After that this shard doesn't show in ```neofs-cli control shards list``` command:
```
root@az:/etc/neofs/storage# sudo neofs-cli --endpoint 127.0.0.1:8091 -w /etc/neofs/storage/wallet.json control shards list
Enter password >
Shard TjX9Vi2EQVmbgox5J78MAj:
Mode: read-write
Metabase: /srv/neofs/meta0/metabase1.db
Blobstor: /srv/neofs/data1
Write-cache: /srv/neofs/meta0/write_cache1
Pilorama: /srv/neofs/meta0/pilorama1.db
Error count: 0
Shard CjzxhoGeYXcowa5nJx5Qno:
Mode: read-write
Metabase: /srv/neofs/meta0/metabase2.db
Blobstor: /srv/neofs/data2
Write-cache: /srv/neofs/meta0/write_cache2
Pilorama: /srv/neofs/meta0/pilorama2.db
Error count: 0
Shard 4HJDfhzxSycFH8xqqD7RcC:
Mode: read-write
Metabase: /srv/neofs/meta1/metabase3.db
Blobstor: /srv/neofs/data3
Write-cache: /srv/neofs/meta1/write_cache3
Pilorama: /srv/neofs/meta1/pilorama3.db
Error count: 0
```
**Config:** [config (3).zip](https://github.com/nspcc-dev/neofs-node/files/9735771/config.3.zip)
**Logs:** [shard_err2.zip](https://github.com/nspcc-dev/neofs-node/files/9735788/shard_err2.zip)

Versions:
```
NeoFS Storage node
Version: v0.32.0-73-g01ddb3f8
GoVersion: go1.18.4
```
## Your Environment
Server setup and configuration:
cloud, 4 VMs, 4 SN, 4 http qw, 4 s3 gw

Operating System and version (uname -a):
linux vedi 5.10.0-16-amd64 https://github.com/nspcc-dev/neofs-node/issues/1 SMP Debian 5.10.127-1 (2022-06-30) x86_64 GNU/Linux

**Comment by fyrchik** [comment]
The service was restarted because of panic:
```
Oct 07 16:43:33 az neofs-node[19293]: unexpected fault address 0x7f19f5da7008
Oct 07 16:43:33 az neofs-node[19293]: fatal error: fault
Oct 07 16:43:33 az neofs-node[19293]: [signal SIGBUS: bus error code=0x2 addr=0x7f19f5da7008 pc=0xbedf31]
Oct 07 16:43:33 az neofs-node[19293]: goroutine 3353 [running]:
Oct 07 16:43:33 az neofs-node[19293]: runtime.throw({0x10c15a2?, 0x7f1a1cd0c108?})
Oct 07 16:43:33 az neofs-node[19293]:         runtime/panic.go:992 +0x71 fp=0xc000bf56f0 sp=0xc000bf56c0 pc=0x438bf1
Oct 07 16:43:33 az neofs-node[19293]: runtime.sigpanic()
Oct 07 16:43:33 az neofs-node[19293]:         runtime/signal_unix.go:815 +0x125 fp=0xc000bf5740 sp=0xc000bf56f0 pc=0x44ed85
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*elemRef).isLeaf(...)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/cursor.go:387
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*Cursor).first(0xc000bf5850)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/cursor.go:170 +0x51 fp=0xc000bf57a0 sp=0xc000bf5740 pc=0xbedf31
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*Cursor).First(0xc000bf5850)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/cursor.go:36 +0x10d fp=0xc000bf5820 sp=0xc000bf57a0 pc=0xbed7ad
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*Bucket).ForEach(0xc0006828d8, 0xc000bf5890)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/bucket.go:389 +0x70 fp=0xc000bf5880 sp=0xc000bf5820 pc=0xbebdb0
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*Tx).ForEach(0x1?, 0xc0006828c0?)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/tx.go:128 +0x49 fp=0xc000bf58b8 sp=0xc000bf5880 pc=0xbfac29
Oct 07 16:43:33 az neofs-node[19293]: github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobovnicza.(*Blobovnicza).Get.func1(0xc0014ffe00?)
Oct 07 16:43:33 az neofs-node[19293]:         github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobovnicza/get.go:49 +0x65 fp=0xc000bf5900 sp=0xc000bf58b8 pc=0xc00485
Oct 07 16:43:33 az neofs-node[19293]: go.etcd.io/bbolt.(*DB).View(0xd4b3887d3da23944?, 0xc000bf59f0)
Oct 07 16:43:33 az neofs-node[19293]:         go.etcd.io/bbolt@v1.3.6/db.go:772 +0x82 fp=0xc000bf5968 sp=0xc000bf5900 pc=0xbf1842
Oct 07 16:43:33 az neofs-node[19293]: github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobovnicza.(*Blobovnicza).Get(0xc001c42fa0, {{{0x44, 0x39, 0xa2, 0x3d, 0x7d, 0x88, 0xb3, 0xd4, 0x2f, ...}, ...}})
```

Because of this the shard was dropped after restart:
```
Oct 07 16:43:40 az neofs-node[19329]: 2022-10-07T16:43:40.039Z        info        writecache/init.go:61        finished updating flush marks        {"shard_id": "CjzxhoGeYXcowa5nJx5Qno"}
Oct 07 16:43:40 az neofs-node[19329]: 2022-10-07T16:43:40.041Z        error        engine/control.go:79        shard initialization failure, skipping        {"id": "Jj5YJnPQTXHSoYg6KpUuQi", "error": "could not initialize *blobstor.BlobStor: failure on blobovnicza initialization stage: could not open blobovnicza 0/2: invalid database"}
```

@anikeev-yadro could you elaborate on the process of READ errors injection? (config for unreliablefs?)

**Comment by anikeev-yadro** [comment]
From the customer side I expect that all shards will be showed in list, but one bad shard will be shown with something bad status, like "Not initialized". It will help administrator to clarify what happined.

**Comment by roman-khimov** [comment]
The node may fail to start at all, shard is not available -> configuration is invalid.

**Comment by roman-khimov** [comment]
Solved by #2464, see #2471 as well.
